codeserver/ubi9-python-3.12/Dockerfile.cpu

```diff
@@ -138,18 +204,28 @@ ENV SHELL=/bin/bash
 
 ENV PYTHONPATH=/opt/app-root/bin/python3
 
-USER 1001
-
 # Install useful packages from requirements.txt
 COPY ${CODESERVER_SOURCE_CODE}/pylock.toml ./
 
+# wait for whl-cache stage (builds uv cache)
+COPY --from=whl-cache /tmp/control /dev/null
+
 # Install packages and cleanup
-RUN echo "Installing softwares and packages" && \
-    # This may have to download and compile some dependencies, and as we don't lock requirements from `build-system.requires`,
-    #  we often don't know the correct hashes and `--require-hashes` would therefore fail on non amd64, where building is common.
-    uv pip install --strict --no-deps --no-cache --no-config --no-progress --verify-hashes --compile-bytecode --index-strategy=unsafe-best-match --requirements=./pylock.toml && \
-    # Fix permissions to support pip in Openshift environments \
-    chmod -R g+w /opt/app-root/lib/python3.12/site-packages && \
+# install packages as USER 0 (this will allow us to consume uv cache)
+RUN --mount=type=cache,target=/root/.cache/uv \
+    echo "Installing softwares and packages" && \
+    # we can ensure wheels are consumed from the cache only by restricting internet access for uv install with '--offline' flag
+    uv pip install --offline --cache-dir /root/.cache/uv --requirements=./pylock.toml && \
```

Comment:
@coderabbitai create github issue for this; is this the way to separate a build and install stage with uv?